### ISSUE TITLE
Server Owners Should Be Able to Have Admin Rights on Any Room and Block More Rooms from Being Created

### ISSUE THREAD

**Issue by BadassOverlord** [matrix-org/synapse#1970]
Hello,

I just started hosting my own server, and I'm using Riot.im for the front end (also hosted on my own server). I was looking to host this as a communications app for my small group of friends (similar to discord), but I was unable to find any Admin options.

Basically, anyone that creates an account on my homeserver can spam create rooms. I don't want anyone except for a select few groups (basically my moderators) to be able to create rooms. I was also looking for Server Owners to be able to get admin on any room, regardless of creation status.

In summary, I guess it could be interpreted that I'm looking for global server groups, because I'm not really seeing the implementation here.

I've been waiting ages for a self-hosted alternative to Discord, and I thought I'd find one with Matrix, but it's really not happening for our group as it stands.

I'd really love for this feature to be implemented!

Thanks for reading.

**Comment by jfrederickson** [comment]
>I was also looking for Server Owners to be able to get admin (op 100) on any room, regardless of creation status.

This can't be done (intentionally) for normal federated rooms, as such rooms aren't owned by a single server. It could maybe be an option for non-federated rooms.

There is [a proposal for server groups](https://docs.google.com/document/d/1cTK2pKolWNXspL69knpDJkcQWZsHpsMDTc2X_dEB5XQ/edit).

**Comment by hamber-dick** [comment]
I don't think that this is a goog idea. This is also a privacy issue. Nobody should have controll over all rooms, even when they are not federated and even when the Person ist the owner of the server.
The only Posibillity i could imagine is some kind of Meta-Admin-Rights. For example the possibillity to delete Rooms. For example because the Room is Dead and there is no more traffic inside. Or a Possibillity for Users to invite the admin into a room for reasons...(Spam, abusive content, racissm...)

**Comment by BadassOverlord** [comment]
Thanks for the responses.

I believe that it does make sense for non-federated servers, but not so much for federated ones. 

As for privacy concerns, I'm not really seeing one. Can you give some examples of what you mean? I'm not too familiar of the privileges that room owners have over normal users.

**Comment by hamber-dick** [comment]
A Private Room with no access, for not invited people, wich is not listed in any directory should stay private even for the Admin.
Or do you want to stork the Users on your HS. Encryption is not suported by every App fo Matrix so the Admin would be able, on a very easy way, to read every chat even the privare ones. I think there should not be a easy way to do this.

**Comment by BadassOverlord** [comment]
I agree that an admin shouldn't be able to see the communications inside the private room, but still should be able to allow/deny creation of private rooms or not depending on server group, and delete the room or rename the room if they need to.

Not all server admins enforce the same rules, and there may even be company policies they have to follow.

**Comment by jfrederickson** [comment]
>delete the room

This would probably be a good thing to have - more specifically, blocking a given room ID from your homeserver (for e.g. if a member of that room reported consistent abuse).

>rename the room

This is already possible - server admins have control over room aliases.

**Comment by richvdh** [comment]
>  I don't want anyone except for a select few groups (basically my moderators) to be able to create rooms.

Is the problem actually with creating rooms, or is it more about who is allowed to create aliases for rooms? The `alias_creation_rules` allow you to lock the latter down these days.
